### PR TITLE
Problem (Fix #1784): possible panic of operations in mls modules

### LINKED ISSUE #1784: Problem: possible panic of operations in mls modules
As TDBE will use the mls library as its part when processing data obtained from the light client, it shouldn't crash e.g. from malformed encryption etc.

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
    - better error hierarchy
    - convert panic operation to error propogation
    - better comment on internal invariants panic

> `astree.rs` is not touched in this pr.

### CODE REVIEW

File: chain-tx-enclave-next/mls/src/ciphersuite.rs [hunk 1]
```diff
@@ -233,20 +229,17 @@ impl CipherSuite {
                 let tag_bytes = &encrypted_group_secret.encrypted_group_secrets.ciphertext
                     [payload_len - 16..payload_len];
```
[the remark this answers is not in the document]
Comment (yihuang) [reply chain-tx-enclave-next/mls/src/ciphersuite.rs:230]: fixed

**Comment by tomtau** [comment]
a small clippy error:
https://travis-ci.org/github/crypto-com/chain/jobs/713138206#L1732

+ compile time error:
```
error[E0308]: mismatched types
1425	   --> chain-tx-enclave-next/mls/src/keypackage.rs:290:35
1426	    |
1427	290 |             KeyPackage { payload, signature },
1428	    |                                   ^^^^^^^^^ expected struct `std::vec::Vec`, found enum `std::result::Result`
1429	    |
1430	    = note: expected struct `std::vec::Vec<_>`
1431	                 found enum `std::result::Result<std::vec::Vec<_>, ring::error::Unspecified>`
1432	
1433	error: aborting due to previous error
1434	
```

**Comment by yihuang** [comment]
> a small clippy error:
> https://travis-ci.org/github/crypto-com/chain/jobs/713138206#L1732
> 
> + compile time error:
> 
> ```
> error[E0308]: mismatched types
> 1425	   --> chain-tx-enclave-next/mls/src/keypackage.rs:290:35
> 1426	    |
> 1427	290 |             KeyPackage { payload, signature },
> 1428	    |                                   ^^^^^^^^^ expected struct `std::vec::Vec`, found enum `std::result::Result`
> 1429	    |
> 1430	    = note: expected struct `std::vec::Vec<_>`
> 1431	                 found enum `std::result::Result<std::vec::Vec<_>, ring::error::Unspecified>`
> 1432	
> 1433	error: aborting due to previous error
> 1434	
> ```

fixed now

**Comment by tomtau** [comment]
bors retry

File: chain-tx-enclave-next/mls/Cargo.toml [hunk 2]
```diff
@@ -11,11 +11,12 @@ ring = "0.16.15"
 thiserror = "1.0"
 rustls = "0.18"
 x509-parser = "0.8.0-beta4"
+nom = "5.1"
```
Comment (tomtau) [inline chain-tx-enclave-next/mls/Cargo.toml:14]: why is nom needed?

Comment (yihuang) [reply chain-tx-enclave-next/mls/Cargo.toml:14]: need the `nom::Err` error type to embed into the error enum.

Comment (tomtau) [reply chain-tx-enclave-next/mls/Cargo.toml:14]: oh, `InvalidCertificate`

**Comment by tomtau** [comment]
bors retry

**Pull request merged**